src/memo/memo.ml

```diff
@@ -302,15 +302,20 @@ module M = struct
 
   (* Why do we store a [run] in the [Considering] state?
 
-     Nodes can currently be invalidated and a build run restarted while some
-     computations are still under way (are in the [Considering] state). We could
-     enforce a separation between the computation and invalidation phases, which
-     could make the implementation simpler, but for now we allow the phases to
-     partially overlap. To distinguish "current" computations from "stale" ones,
-     each computation stores the [run] in which it was started. In this way,
-     before subscribing to an [Ivar.t] from the [completion], we can check if it
-     corresponds to the "current" run, and if not, the computation should be
-     restarted. *)
+     It used to be possible for a computation to finish with a cycle error and
+     remain in the [Considering] state forever, becoming a "zombie" computation.
+
+     To distinguish between "current" and "zombie" computations, we stored the
+     [run] in which the computation had started. In this way, before subscribing
+     to an [Ivar.t] from the [completion], we could check if it corresponded to
+     the current run, and if not, start a new computation.
+
+     With a recent (as of 2021-02-18) refactoring, we believe such zombie
```

Comment:
FWIW, If we're to attribute the removal of zombies to any specific refactoring, I think that would be the Rudi's change to explicitly handle errors, which pretty much eliminated zombies except perhaps in some weird cases.

Reply:
You are right, it turns out it was @rgrinberg who freed us from zombies. I've updated the dates.